crates/youki/src/commands/features.rs

```diff
@@ -1,8 +1,203 @@
 //! Contains Functionality of `features` container command
+use std::path::Path;
+
 use anyhow::Result;
+use caps::{all, CapSet};
+use libcontainer::oci_spec::runtime::{
+    version, ApparmorBuilder, CgroupBuilder, FeaturesBuilder, IDMapBuilder, IntelRdtBuilder,
+    LinuxFeatureBuilder, LinuxNamespaceType, MountExtensionsBuilder, SelinuxBuilder,
+};
 use liboci_cli::Features;
 
+// Function to query and return capabilities
+fn query_caps() -> Result<Vec<String>> {
+    let mut available_caps = Vec::new();
+
+    for cap in all() {
+        // Check if the capability is in the permitted set
+        if caps::has_cap(None, CapSet::Permitted, cap).unwrap_or(false) {
+            available_caps.push(format!("{:?}", cap));
+        }
+    }
+
+    Ok(available_caps)
+}
+
+// Function to query and return namespaces
+fn query_supported_namespaces() -> Result<Vec<LinuxNamespaceType>> {
+    let mut supported_namespaces = Vec::new();
+
+    let ns_types = vec![
+        LinuxNamespaceType::Pid,
+        LinuxNamespaceType::Network,
+        LinuxNamespaceType::Uts,
+        LinuxNamespaceType::Ipc,
+        LinuxNamespaceType::Mount,
+        LinuxNamespaceType::User,
+        LinuxNamespaceType::Cgroup,
+        LinuxNamespaceType::Time,
+    ];
+
+    for ns in ns_types {
+        let ns_path = format!("/proc/self/ns/{}", ns);
+        if Path::new(&ns_path).exists() {
+            supported_namespaces.push(ns);
+        }
+    }
+
+    Ok(supported_namespaces)
+}
+
+fn known_hooks() -> Vec<String> {
+    vec![
+        String::from("prestart"),
+        String::from("createRuntime"),
+        String::from("createContainer"),
+        String::from("startContainer"),
+        String::from("poststart"),
+        String::from("poststop"),
+    ]
+}
+
+fn known_mount_options() -> Vec<String> {
+    vec![
+        String::from("acl"),
+        String::from("async"),
+        String::from("atime"),
+        String::from("bind"),
+        String::from("defaults"),
+        String::from("dev"),
+        String::from("diratime"),
```

Comment:
I feel a better way to do this, especially in case of `mount_options` is 
```rust
["acl","async",...].into_iter().map(String::from).collect()
```
This way we don't  have to repeat the `String::from` every time.

Also, does youki actually support all these? I don't think I have seen the acl or lazytime anywhere in mount code. I think this should  have only the options that we actually support (i.e. options that `nix` crate actually allows us to set)

Reply:
Let me check.

Reply:
I've fixed based on your suggestion in 79f389e, but let me check if each of them is supported in youki.

Reply:
I'll make a fix for this part based on the actual implementation.

https://github.com/musaprg/youki/blob/79f389e735e4a368b348f671a2b2761dd184424f/crates/libcontainer/src/rootfs/utils.rs#L85-L191

Reply:
fixed in eb5cd7e...964755c